src/screens/Routes/onNavigationStateChange.ios.js

```diff
@@ -0,0 +1,118 @@
+import analytics from '@segment/analytics-react-native';
+import { get } from 'lodash';
+import { StatusBar } from 'react-native';
+import { NavigationActions } from 'react-navigation';
+import isNativeStackAvailable from '../../helpers/isNativeStackAvailable';
+import { Navigation } from '../../navigation';
+import { expandedPreset } from '../../navigation/transitions/effects';
+import { sentryUtils } from '../../utils';
+import { ROUTES } from './routesNames';
+
+const routesForNativeStackFallback = {
+  [ROUTES.ADD_CASH_SHEET]: true,
+  [ROUTES.IMPORT_SEED_PHRASE_SHEET]: true,
+  [ROUTES.MAIN_NAVIGATOR]: true,
+  [ROUTES.OVERLAY_EXPANDED_ASSET_SCREEN]: true,
+  [ROUTES.OVERLAY_EXPANDED_SUPPORTED_COUNTRIES]: true,
+  [ROUTES.SEND_SHEET]: true,
+  [ROUTES.MAIN_NATIVE_BOTTOM_SHEET_NAVIGATOR]: true,
+  [ROUTES.SAVINGS_DEPOSIT_MODAL]: true,
+  [ROUTES.SAVINGS_DEPOSIT_MODAL]: true,
+};
+
+export function onNavigationStateChange(prevState, currentState) {
+  const { params, routeName } = Navigation.getActiveRoute(currentState);
+  const prevRouteName = Navigation.getActiveRouteName(prevState);
+  // native stack rn does not support onTransitionEnd and onTransitionStart
+  // Set focus manually on route changes
+
+  if (
+    prevRouteName !== routeName &&
+    isNativeStackAvailable &&
+    (routesForNativeStackFallback[prevRouteName] ||
+      routesForNativeStackFallback[routeName])
+  ) {
+    Navigation.handleAction(
+      NavigationActions.setParams({
+        key: routeName,
+        params: { focused: true },
+      })
+    );
+
+    Navigation.handleAction(
+      NavigationActions.setParams({
+        key: prevRouteName,
+        params: { focused: false },
+      })
+    );
+  }
+
+  const oldBottomSheetStackindex = prevState.index;
+  const newBottomSheetStackIndex = currentState.index;
+
+  if (oldBottomSheetStackindex !== newBottomSheetStackIndex) {
+    expandedPreset.onTransitionStart({ closing: !newBottomSheetStackIndex });
+  }
+
+  if (
+    prevRouteName !== ROUTES.QR_SCANNER_SCREEN &&
+    routeName === ROUTES.QR_SCANNER_SCREEN
```

Comment:
Basically, these sections repeat the `.android.js` version and should be unified before merging. If we move information about side-effects (like status bar color changes) to the route parameters, then the route change handler function will be pretty tiny. 

Now, after looking at these two files, I wonder what's the reason to split them up?

Reply:
It was just following up of having it in `Routes.ios.js` and `.android.js`